trace/workflow_state_worker.go

```diff
@@ -0,0 +1,139 @@
+package trace
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/alitto/pond"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/history/v1"
+	sdkclient "go.temporal.io/sdk/client"
+)
+
+// WorkflowStateJob implements a WorkerJob to retrieve updates for a WorkflowExecutionState and its child workflows.
+type WorkflowStateJob struct {
+	ctx    context.Context
+	client sdkclient.Client
+
+	state      *WorkflowExecutionState
+	depth      int
+	fetchAll   bool
+	childJobs  []*WorkflowStateJob
+	isUpToDate bool
+
+	updateChan chan struct{}
+}
+
+// NewWorkflowStateJob returns a new WorkflowStateJob. It requires an updateChan to signal when there's updates.
+func NewWorkflowStateJob(ctx context.Context, client sdkclient.Client, state *WorkflowExecutionState, fetchAll bool, depth int, updateChan chan struct{}) (*WorkflowStateJob, error) {
+	if state == nil {
+		return nil, errors.New("workflow state cannot be nil for a workflow state job")
+	}
+	if updateChan == nil {
+		return nil, errors.New("updateChan cannot be nil for a workflow state job")
+	}
+
+	// Get workflow execution's description, so we can know if we're up-to-date. Doing this synchronously will allow us to correctly
+	// assess how many events need to be processed (otherwise only the ones from the root workflow will be counted).
+	// We don't mind if this fails, since HistoryLength is used only to display event processing progress.
```

Comment:
I'd rather just return the error here instead of silently ignoring it. There's no good reason for this call to fail and if it does it'll be hard to debug.

Reply:
I disagree, but I'll update it as required.